src/Kafka/Internal/Shared.hs

```diff
@@ -13,13 +13,14 @@ import           Kafka.Internal.Setup
 import           Kafka.Types
 
 runEventLoop :: HasKafka a => a -> CancellationToken -> Maybe Timeout -> IO ()
-runEventLoop k ct timeout = void $ forkIO go
-  where
-    go = do
-      token <- CToken.status ct
-      case token of
-        Running   -> pollEvents k timeout >> go
-        Cancelled -> return ()
+runEventLoop k ct timeout =
+    when rtsSupportsBoundThreads $ void $ forkIO go
```

Comment:
This is the real change: runs the event loop in another IO thread only if RTS is capable of doing parallel things

Reply:
I test it and it works with both threaded and non-threaded. When `threaded` is disabled `runEventLoop` does nothing. I don't figure out how could `newProducer`/`newConsumer`  run normally as before. Could you please explain https://github.com/haskell-works/hw-kafka-client/blob/master/src/Kafka/Producer.hs#L70 to me? Thanks!